org.eclipse.egit.core/src/org/eclipse/egit/core/Activator.java

```diff
@@ -248,10 +248,20 @@ public void optionsChanged(DebugOptions options) {
 	 * @since 4.1
 	 */
 	public MergeStrategy getPreferredMergeStrategy() {
+		// Get preferences set by user in the UI
 		final IEclipsePreferences prefs = InstanceScope.INSTANCE
 				.getNode(Activator.getPluginId());
 		String preferredMergeStrategyKey = prefs.get(
 				GitCorePreferences.core_preferredMergeStrategy, null);
+
+		// Get preferences defined at eclipse startup with configuration files
+		if (preferredMergeStrategyKey == null
```

Comment:
In EMFCompare the class ScopedPreferenceStore is used to search a specific preference across multiple preferenceStores. This approach seems to be good for managing EGit preferences, but the preferences are managed in egit.core and ScopedPreferenceStore comes from eclipse.ui.

I think the better approach is to manage each store individually, as EGit does